src/main/java/io/r2dbc/postgresql/codec/BuiltinDynamicCodecs.java

```diff
@@ -48,6 +49,8 @@
             switch (this) {
                 case HSTORE:
                     return new HStoreCodec(byteBufAllocator, oid);
+                case POSTGIS_GEOMETRY:
```

Comment:
The code should include a guard to verify that the required classes for `jts-core` are on the class path.

Reply:
Added `isSupported` method and changed to register codec only when it returns `true`.